CMakeLists.txt

```diff
@@ -8,31 +8,45 @@ project(Tracy LANGUAGES CXX VERSION ${TRACY_VERSION_STRING})
 file(GENERATE OUTPUT .gitignore CONTENT "*")
 
 if(${BUILD_SHARED_LIBS})
-	set(DEFAULT_STATIC OFF)
+    set(DEFAULT_STATIC OFF)
 else()
-	set(DEFAULT_STATIC ON)
+    set(DEFAULT_STATIC ON)
 endif()
 
 option(TRACY_STATIC "Whether to build Tracy as a static library" ${DEFAULT_STATIC})
 option(TRACY_Fortran "Build Fortran bindings" OFF)
+option(TRACY_LTO "Enable Link-Time optimization" OFF)
 
 if(TRACY_Fortran)
   enable_language(Fortran)
   set(CMAKE_Fortran_VERSION 2003)
 endif()
 
+if(TRACY_LTO)
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT LTO_SUPPORTED)
+    if(NOT LTO_SUPPORTED)
+        message(WARNING "LTO is not supported!")
+    endif()
+else()
+    set(LTO_SUPPORTED OFF)
+endif()
+
 find_package(Threads REQUIRED)
 
 set(TRACY_PUBLIC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/public)
 
-if(TRACY_STATIC)
-	set(TRACY_VISIBILITY "STATIC")
+if(LTO_SUPPORTED)
+    set(TRACY_VISIBILITY "OBJECT")
+elseif(TRACY_STATIC)
+    set(TRACY_VISIBILITY "STATIC")
 else()
-	set(TRACY_VISIBILITY "SHARED")
+    set(TRACY_VISIBILITY "SHARED")
 endif()
 
 add_library(TracyClient ${TRACY_VISIBILITY} "${TRACY_PUBLIC_DIR}/TracyClient.cpp")
 target_compile_features(TracyClient PUBLIC cxx_std_11)
+set_target_properties(TracyClient PROPERTIES INTERPROCEDURAL_OPTIMIZATION ${LTO_SUPPORTED})
```

Comment:
This would interfere with the global option.

Reply:
Fixed by https://github.com/wolfpld/tracy/pull/967/commits/b87a935c58a89ec9f7b73b5a3c88b7f9b7f23fbe